lib/iris/_lazy_data.py

```diff
@@ -23,12 +23,14 @@
 from __future__ import (absolute_import, division, print_function)
 from six.moves import (filter, input, map, range, zip)  # noqa
 
+from collections import Iterable
```

Comment:
@pp-mo See https://github.com/SciTools/iris/pull/3320 for context.

Currently in Python3.7, you get the following `DeprecationWarning`:
```python
>>> from collections import Iterable
__main__:1: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
```
Could you adopt the following pattern:
```python
try:  # Python 3
    from collections.abc import Iterable
except ImportError:  # Python 2
    from collections import Iterable
```

Reply:
Looks good, will do !